src/containers/itemcollection/index.js

```diff
@@ -0,0 +1,147 @@
+import React, { Component } from 'react'
+
+import { connect } from 'react-redux'
+
+import PropTypes from 'prop-types'
+import { Grid, Typography } from '@material-ui/core'
+
+import Item from 'components/item'
+import EditItem from 'components/edititem'
+
+import { addItem } from 'actions/item'
+
+export class ItemCollection extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      isEdit: false,
+      name: '',
+      _id: '',
+      size: 1
+    }
+
+    this.addEditItem = this.addEditItem.bind(this)
+    this.displayEditItem = this.displayEditItem.bind(this)
+    this.handleEditItemChange = this.handleEditItemChange.bind(this)
+    this.handleEditItemEnterKey = this.handleEditItemEnterKey.bind(this)
+    this.handleEditItemEscKey = this.handleEditItemEscKey.bind(this)
+  }
+
+  addEditItem () {
+    this.setState({
+      isEdit: true,
+      _id: this.uuid(),
+      name: '',
+      size: 1
+    })
+  }
+
+  uuid() {
+    var text = "i";
+    var possible = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+  
+    for (var i = 0; i < 8; i++)
+      text += possible.charAt(Math.floor(Math.random() * possible.length));
+  
+    return text;
+  }
+
+  handleEditItemChange (e) {
+    this.setState({
+      ...this.state,
+      name: e.target.value
+    })
+  }
+
+  handleEditItemEnterKey () {
+    const item = {
+      _id: this.state._id,
+      name: this.state.name,
+      size: this.state.size
+    }
+
+    this.setState({
+      isEdit: false,
+      name: '',
+      _id: '',
+      size: 1
```

Comment:
What is this size for?

Reply:
this is the size of the item, which we assume is size 1.